src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
```

Comment:
The function `groupByKey` assumes that `key` will always be a property of `T` that results in a string when used as an index. If `T` has a property that is not a string, this could lead to a runtime error. Consider adding a constraint to `T` to ensure `key` is a property whose values are assignable to string keys.